src/wasm-lib/grackle/src/native_functions/sketch/stdlib_functions.rs

```diff
@@ -171,26 +280,88 @@ impl Callable for LineTo {
                 EpBinding::Single(empty_string_addr)
             }
         };
+
         // Check the type of required params.
-        let to = arg_point2d(to, fn_name, &mut instructions, ctx, 0)?;
+        // We don't check `to` here because it can take on either a
+        // EpBinding::Sequence or EpBinding::Single.
+
         let sg = sg_binding(sketch_group, fn_name, "sketch group", 1)?;
         let tag = single_binding(tag, fn_name, "string tag", 2)?;
         let id = Uuid::new_v4();
+
         // Start of the path segment (which is a straight line).
         let length_of_3d_point = Point3d::<f64>::default().into_parts().len();
         let start_of_line = ctx.next_address.offset_by(1);
+
+        // Reserve space for the segment last point
+        let to_point_from = ctx.next_address.offset_by(2);
+
         // Reserve space for the line's end, and the `relative: bool` field.
         ctx.next_address.offset_by(length_of_3d_point + 1);
         let new_sg_index = ctx.assign_sketch_group();
+
+        // Copy based on the options.
+        match opts {
+            LineBareOptions { at: At::AbsoluteXY, .. } | LineBareOptions { at: At::RelativeXY, .. } => {
+                // Push the `to` 2D point onto the stack.
+                if let EpBinding::Sequence { elements, length_at: _ } = to.clone() {
+                    if let &[EpBinding::Single(el0), EpBinding::Single(el1)] = elements.as_slice() {
+                        instructions.extend([
+                            Instruction::Copy {
+                                source: el0,
+                                length: 1,
+                                destination: Destination::StackPush,
+                            },
+                            Instruction::Copy {
+                                source: el1,
+                                length: 1,
+                                destination: Destination::StackExtend,
+                            },
+                            // Make it a 3D point.
+                            Instruction::StackExtend { data: vec![0.0.into()] },
+                        ]);
+                    } else {
+                        panic!("Must pass a list of length 2");
```

Comment:
Unnesting the `if`s and making them `let-else` blocks instead will make this more readable IMO. Also returning an error instead of panicking will be a better UX (as I suspect you were planning to do anyway).